### ISSUE TITLE
Implement v8::Isolate::SetAbortOnUncaughtExceptionCallback

### ISSUE THREAD

**Issue by brendandahl** [mozilla/spidernode#126]
Node's use: support the command line flag `--abort-on-uncaught-exception`. There's some logic to handle different abort behavior when sub processes are run and if they have their own top level error handlers. https://github.com/nodejs/node/commit/77a10ed05f1e413818d29f07cbb268108b1f08fa For node users this is mainly to generate a core dump when the flag is specified.

Chakra: stubbed out

V8: the callback and flag is handled within v8's isolate implementation of Throw(), which in turn is triggered by Isolate:ThrowException.

To implement it looks like we'd need to expand on the work in #124 and if the exception isn't caught and the callback return `true` then we need to abort.

**Comment by brendandahl** [comment]
I don't think we can currently implement this easily. In V8, it's possible to check if the current exception will be caught by JS, which allows node to abort right away and produce a core dump. In SM, there's a way to check if the exception will be caught using the debugger api (`isInCatchScope`), but this isn't exposed in through jsapi. We could run some JS to do this, but that seems rather hacky.

**Comment by ehsan** [comment]
Hmm, can you please clarify what you mean by "will be caught by JS"?  Does that mean if we're inside a try block?  Or if there is a try block on the call stack somewhere?  Or something else?

Is this something that we can ask help with from the SpiderMonkey team?

**Comment by brendandahl** [comment]
Yes, inside a try block. V8's relevant code https://github.com/v8/v8/blob/49411f5f0585a32715d1f1883e175f7bd69dd375/src/isolate.cc#L1251

> Is this something that we can ask help with from the SpiderMonkey team?

Yeah, this seems on the lower priority side of things since it's mainly used for debugging. Should I file an upstream issue?

**Comment by ehsan** [comment]
(Sorry for the delay.)

> Yes, inside a try block. V8's relevant code https://github.com/v8/v8/blob/49411f5f0585a32715d1f1883e175f7bd69dd375/src/isolate.cc#L1251

Interesting.

I wonder whether we can use our bytecode information to do something similar.  Worth checking with the JS team on how to do that.

> ```
> Is this something that we can ask help with from the SpiderMonkey team?
> ```
> 
> Yeah, this seems on the lower priority side of things since it's mainly used for debugging. Should I file an upstream issue?

Yes, please.  :-)
